// my-react-app/src/theme/colors.js
export const colors = {
  // Background colors
  background: {
    primary: '#0f172a',    // Dark blue background
    secondary: '#1e293b',  // Slightly lighter blue for cards/containers
    tertiary: '#121a2b',   // Darker blue for nav/footers
  },
  
  // Text colors
  text: {
    primary: '#ffffff',    // White for main text
    secondary: '#94a3b8',  // Light gray for secondary text
    muted: '#64748b',      // Muted text for placeholders
  },
  
  // UI Colors
  ui: {
    primary: '#4f46e5',     // Primary button/action color
    primaryHover: '#4338ca',  // Hover state for primary
    border: '#334155',       // Border color
    inputBg: '#0f172a',      // Input background
  },
  
  // State colors
  state: {
    disabled: '#4b5563',   // Disabled state
    error: '#ef4444',       // Error messages
    success: '#10b981',     // Success messages
    warning: '#f59e0b',     // Warning messages
  },
  
  // Additional colors
  additional: {
    shadow: 'rgba(0, 0, 0, 0.1)', // Shadow color
    overlay: 'rgba(0, 0, 0, 0.5)',  // Overlay/modal background
  }
};

export default colors;
